superset/views/core.py

```diff
@@ -1831,15 +1858,15 @@ def publish(self, dashboard_id):
             return json_success(json.dumps({"published": dash.published}))
 
     @has_access
-    @expose("/dashboard/<dashboard_id>/")
-    def dashboard(self, dashboard_id):
+    @expose("/dashboard/<dashboard_id_or_slug>/")
```

Comment:
`dashboard_id_or_slug` seemed to be a more accurate representation of the field.